grpc-bridge/src/main/java/org/jboss/resteasy/grpc/protobuf/JavabufTranslatorGenerator.java

```diff
@@ -0,0 +1,556 @@
+           .append("      static {\n")
+           .append("         for (FieldDescriptor f : descriptor.getFields()) {\n")
+           .append("            String name = f.getName();\n")
+           .append("            if (descriptor.findFieldByName(name) == null) {\n")
+           .append("               continue;\n")
+           .append("            }\n")
+           .append("            assignList.add(fromJavabuf(").append(originalName).append(".class, descriptor.findFieldByName(name)));\n")
+           .append("         }\n")
+           .append("      }\n\n")
+           .append("      public ").append(originalName).append(" assignFromJavabuf(Message message) {\n")
+           .append("         ").append(originalName).append(" obj = new ").append(originalName).append("();\n")
+           .append("         for (AssignFromJavabuf assignFrom : assignList) {\n")
+           .append("            try {\n")
+           .append("               assignFrom.assign(message, obj);\n")
+           .append("            } catch (Exception e) {\n")
+           .append("               throw new RuntimeException(e);\n")
+           .append("            }\n")
+           .append("         }\n")
+           .append("         return obj;\n")
+           .append("      }\n\n")
+           .append("      public void assignExistingFromJavabuf(Message message, Object obj) {\n")
+           .append("         for (AssignFromJavabuf assignFrom : assignList) {\n")
+           .append("            try {\n")
+           .append("               assignFrom.assign(message, obj);\n")
+           .append("            } catch (Exception e) {\n")
+           .append("               throw new RuntimeException(e);\n")
+           .append("            }\n")
+           .append("         }\n")
+           .append("      }\n");
+      }
+      sb.append("   }\n\n");
+   }
+
+   private static void finishClass(StringBuilder sb) {
+      sb.append("}\n");
+   }
+
+   private static void writeTranslatorClass(String[] args, String translatorClass, StringBuilder sb) throws IOException {
+      StringBuilder root = new StringBuilder("target/generated-sources/protobuf/grpc-java/");
+      File dir = new File(root.toString());
+      if(!dir.exists()) {
+         dir.mkdir();
+      }
+      String pkg = args[1].lastIndexOf(".") < 0 ? "" : args[1].substring(0, args[1].lastIndexOf("."));
+      for (String s : pkg.split("\\.")) {
+         dir = new File(root.append("/").append(s).toString());
+         if (!dir.exists())
+         {
+            dir.mkdir();
+         }
+      }
+      File file = new File(root.append("/").append(translatorClass).append(".java").toString());
+      if (!file.exists()) {
+         file.createNewFile();
+      }
+      FileWriter fw = new FileWriter(file.getAbsoluteFile());
+      BufferedWriter bw = new BufferedWriter(fw);
+      bw.write(sb.toString());
+      bw.close();
```

Comment:
Same as the note above about try-with-resources or using the new file API.

Reply:
Same